### PR TITLE
adding functionality to allow moderators to add eventId and synonyms

### PR DESCRIPTION
**Pull request by Courey**
This is PR 1 of 3. Once both PRs are in, this body of work must be connected to the synonyms table/index. I have noted where and what change needs to be made.

### CODE REVIEW

**Comment by lpsinger** [comment]
Which PR should I look at to see the data access patterns for searching, querying, listing, and retrieving Circulars?

**Comment by Courey** [comment]
> Which PR should I look at to see the data access patterns for searching, querying, listing, and retrieving Circulars?

The next one. I'm trying to sort my branches so they have as few conflicts as possible. I am going to turn this back into a draft because I think I snagged too many files.

**Comment by lpsinger** [comment]
> > Which PR should I look at to see the data access patterns for searching, querying, listing, and retrieving Circulars?
> 
> The next one. I'm trying to sort my branches so they have as few conflicts as possible. I am going to turn this back into a draft because I think I snagged too many files.

OK. I'd like to review that one _first_ because my main concern is that the table structure is viable (i.e., avoided expensive scans).

**Comment by Courey** [comment]
> > > Which PR should I look at to see the data access patterns for searching, querying, listing, and retrieving Circulars?
> > 
> > 
> > The next one. I'm trying to sort my branches so they have as few conflicts as possible. I am going to turn this back into a draft because I think I snagged too many files.
> 
> OK. I'd like to review that one _first_ because my main concern is that the table structure is viable (i.e., avoided expensive scans).

you can look at the entire code changes here. you changed the file structure beneath this PR, so the files will not be the correct files (which is why picking them apart instead of rebasing as rebasing did not work with such drastic differences), but the intent of the code is the same. https://github.com/nasa-gcn/gcn.nasa.gov/commit/2ac4da903e9e7d68edcba3cd8b6d34b5e952698f#

File: app/routes/circulars/circulars.server.ts [hunk 1]
```diff
@@ -263,3 +264,46 @@ export async function circularRedirect(query: string) {
...
+    Key: { circularId },
+    UpdateExpression: 'set #eventId = :eventId, #synonyms = :synonyms',
+    ExpressionAttributeNames: {
+      '#eventId': 'eventId',
+      '#synonyms': 'synonyms',
+    },
+    ExpressionAttributeValues: {
+      ':eventId': eventId,
+      ':synonyms': formatSynonyms(synonymsList),
+    },
```
Comment (Courey) [inline app/routes/circulars/circulars.server.ts:306]: Before the feature flag is turned on, I need to connect the two PRs by calling this function here to sync the synonyms [syncSynonyms](https://github.com/nasa-gcn/gcn.nasa.gov/pull/1505/files#diff-9e61d3077d7bdde9ad98586fcc58c0a7c3d81c7562008bbc7a2ca75eab44d2baR283)

**Pull request closed without merging**